packages/integration-tests/src/EntityManager.clone.test.ts

```diff
@@ -24,6 +36,22 @@ describe("EntityManager.clone", () => {
     expect(p1.authors.get).toEqual([a1, a2]);
   });
 
+  it("can clone m2os and maintain loaded", async () => {
+    // Given an entity created in 1 UoW
+    const em = newEntityManager();
+    const p1 = newPublisher(em, { name: "p1" });
+    const a1 = new Author(em, { firstName: "a1", publisher: p1 });
+    await em.flush();
+
+    // When we clone it in a 2nd UoW
+    const em2 = newEntityManager();
+    const a2 = await em2.load(Author, a1.idOrFail);
+    const a3 = await em2.clone(a2);
+
+    // Then the a3.publisher loaded state is correct
+    expect(a3.publisher.isLoaded).toBe(false);
```

Comment:
The "only call `.load` if `!isLoaded`" optimization to EntityManager broke `M2O.get` for entities that were cloned, b/c the cloned value would:

1) See its `M2O.initializeNewEntity` set `isLoaded` to true
2) But the value dropped into `clone.__orm.data` could end up being a string

Fixing this turned into a large refactoring of `em.clone` to not longer just write directly to `__orm.data`, but to go through `em.create` / the constructor with a set of opts just like any other caller would.

I think this is much cleaner, albeit it required teaching `PolymorphicReference` how to be initialized via `em.create` with non-entity/string ids (similar to how we taught `ManyToOneReference` to do the same a few months ago).